### ISSUE TITLE
Can't uninstall on darwin, fail to remove `nix-darwin`

### ISSUE THREAD

**Issue by mamcx** [DeterminateSystems/nix-installer#607]
I try to start clean because everything looks broken. I follow exactly the steps of the tutorial

I just run the tutorial as-is:

```bash
➜ curl --proto '=https' --tlsv1.2 -sSf -L https://install.determinate.systems/nix | sh -s -- install
info: downloading installer https://install.determinate.systems/nix/tag/v0.11.0/nix-installer-aarch64-darwin
`nix-installer` needs to run as `root`, attempting to escalate now via `sudo`...
```

, then generate the flake `nix flake init -t nix-darwin` and get:

```nix
{
  description = "Example Darwin system flake";

  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
    nix-darwin.url = "github:LnL7/nix-darwin";
    nix-darwin.inputs.nixpkgs.follows = "nixpkgs";
  };

  outputs = inputs@{ self, nix-darwin, nixpkgs }:
  let
    configuration = { pkgs, ... }: {
      # List packages installed in system profile. To search by name, run:
      # $ nix-env -qaP | grep wget
      environment.systemPackages =
        [ pkgs.vim
        ];

      # Auto upgrade nix package and the daemon service.
      services.nix-daemon.enable = true;
      # nix.package = pkgs.nix;

      # Necessary for using flakes on this system.
      nix.settings.experimental-features = "nix-command flakes";

      # Create /etc/zshrc that loads the nix-darwin environment.
      programs.zsh.enable = true;  # default shell on catalina
      # programs.fish.enable = true;

      # Set Git commit hash for darwin-version.
      system.configurationRevision = self.rev or self.dirtyRev or null;

      # Used for backwards compatibility, please read the changelog before changing.
      # $ darwin-rebuild changelog
      system.stateVersion = 4;

      # The platform the configuration will be used on.
      nixpkgs.hostPlatform = "x86_64-darwin";
    };
  in
  {
    # Build darwin flake using:
    # $ darwin-rebuild build --flake .#mandalorian
    darwinConfigurations."mandalorian" = nix-darwin.lib.darwinSystem {
      modules = [ configuration ];
    };

    # Expose the package set, including overlays, for convenience.
    darwinPackages = self.darwinConfigurations."mandalorian".pkgs;
  };
}
```

I do a lot of changes, trying to get things my way, and eventually I ended in a situation where even *renaming * to `_flake.nix` and running `nix run nix-darwin -- switch --flake .#mandalorian` always get the error:

```bash
error: a 'aarch64-darwin' with features {} is required to build '/nix/store/6cdpizb5azlyldywpnrnqwmbhg2glrb7-darwin-rebuild.drv', but I am a 'x86_64-darwin' with features {benchmark, big-parallel, nixos-test}
```

Now, doing `gc`, removing that store, using a totally different `flake.nix` get to the same error.

Then try to uninstall:

```bash
/nix/nix-installer uninstall

`nix-darwin` installation detected, it must be removed before uninstalling Nix. Please refer to https://github.com/LnL7/nix-darwin#uninstalling for instructions how to uninstall `nix-darwin`.
```

Ok, so I follow that instructions:

```bash
❯ nix-build https://github.com/LnL7/nix-darwin/archive/master.tar.gz -A uninstaller
this derivation will be built:
  /nix/store/agc95dgwkqchn40c6qjr2008ry3lpgkl-darwin-uninstaller.drv
error: a 'aarch64-darwin' with features {} is required to build '/nix/store/agc95dgwkqchn40c6qjr2008ry3lpgkl-darwin-uninstaller.drv', but I am a 'x86_64-darwin' with features {benchmark, big-parallel, nixos-test}
```

What I can do?

**Comment by Hoverbear** [comment]
In your config you are specifying you are a an `x86_64-darwin` however you've installed the `aarch64-darwin` version of Nix. I assume you have an `aarch64-darwin` machine so you need to update the ` nixpkgs.hostPlatform = "x86_64-darwin";` line to reflect the different architecture.

Please re-open if this doesn't fix it.

**Comment by pickfire** [comment]
I have the same issue as I accidentally did not change the code to `aarch64-darwin`, when I tried to change it back to `aarch64-darwin` and rebuild, it shows
```
> nix --extra-experimental-features "nix-command flakes" run nix-darwin -- switch --flake .
warning: ignoring untrusted substituter 'https://devenv.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://nixpkgs-python.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://devenv.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://nixpkgs-python.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://devenv.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://nixpkgs-python.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
building the system configuration...
warning: ignoring untrusted substituter 'https://devenv.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://nixpkgs-python.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://devenv.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
warning: ignoring untrusted substituter 'https://nixpkgs-python.cachix.org', you are not a trusted user.
Run `man nix.conf` for more information on the `substituters` configuration option.
error: a 'aarch64-darwin' with features {} is required to build '/nix/store/ffmbby8harwpqa01c721652g23snid80-builder.pl.drv', but I am a 'x86_64-darwin' with features {apple-virt, benchmark, big-parallel, nixos-test}
```

**Comment by modulitos** [comment]
I was blocked on this error as well.

I tried adding `extra-platforms = aarch64-darwin x86_64-darwin` to `/etc/nix/nix.conf`, but I was still getting the error.

I was able to uninstall nix as follows:

1. uninstall `nix-darwin`  with this command: `nix-build https://github.com/LnL7/nix-darwin/archive/master.tar.gz -A uninstaller --option system x86_64-darwin`

It's the same [uninstall command from the docs](http://daiderd.com/nix-darwin/#uninstalling), except that I had to pass in `--option system x86_64-darwin` to allow the build to succeed.

2. Run `./result/bin/darwin-uninstaller` to uninstall `nix-darwin`
3. Uninstall nix via the [DeterminateSystems tool](https://github.com/DeterminateSystems/nix-installer?tab=readme-ov-file#uninstalling) (this took a couple of tries and a reboot)

Now I'm back to having a clean Nix install on MacOS, free from the pitfalls of nix-darwin :)
